contracts/voting-verifier/src/events.rs

```diff
@@ -143,6 +143,12 @@ fn parse_message_id(
 
             Ok((id.signature_as_base58(), id.event_index))
         }
+        MessageIdFormat::HexTxHash => {
+            let id = HexTxHash::from_str(&message_id)
+                .map_err(|_| ContractError::InvalidMessageID(message_id.into()))?;
+
+            Ok((id.tx_hash_as_hex(), 0))
```

Comment:
this seems a bit odd to put 0 as the event index here. We should probably refactor the event a bit to not need to include an event index here. Maybe the `TxEventConfirmation` event should have a `tx_info` field that is an enum, which can be different types.

Reply:
That would mean also refactoring ampd and other parts, I don't think that is in the scope of this PR.

Reply:
I think that change is worth doing, but would you prefer to do it in a follow up PR?